sormas-ui/src/main/java/de/symeda/sormas/ui/user/UserEditForm.java

```diff
@@ -195,10 +199,25 @@ protected void addFields() {
         updateFieldsByUserRole();
     }
 
-    @SuppressWarnings("unchecked")
+	@Override
+	public void setValue(UserDto newFieldValue) throws ReadOnlyException, Converter.ConversionException {
+		final OptionGroup userRolesField = (OptionGroup)getFieldGroup().getField(UserDto.USER_ROLES);
+		FieldHelper.updateItems(userRolesField, getFilteredUserRoles(newFieldValue.getUserRoles()));
+
+		super.setValue(newFieldValue);
+	}
+
+	private List<UserRoleReferenceDto> getFilteredUserRoles(Set<UserRoleReferenceDto> selectedUserRoles) {
+		return userRoleMap
+				.entrySet().stream().filter(r -> r.getValue().isEnabled() || selectedUserRoles.stream().anyMatch(s -> DataHelper.isSame(s, r.getValue())))
```

Comment:
```suggestion
				.entrySet().stream().filter(r -> r.getValue().isEnabled() || selectedUserRoles.contains(r.getKey()))
```
This should also work, but I haven't tested it.

Reply:
Tested it and doesn't work